lib/imports/api/authority-registry/server/authority-registry.js

```diff
@@ -0,0 +1,319 @@
+import {
+  insertAuthority,
+  removeAuthority,
+  storeFailedHealthChecks,
+  removeAuthorities,
+  resetFailedHealthChecksCount,
+  addDocToAuthority
+} from './methods';
+import { authorityRegistryColl } from './collection';
+import { WebApp } from 'meteor/webapp';
+import { HTTP } from 'meteor/http';
+import { check } from 'meteor/check';
+import { Meteor } from 'meteor/meteor';
+import ip from 'ip';
+
+const HTTPS_PORT = 443;
+const HEALTH_CHECK_RESPONSE = 'ProseMeteor Health OK';
+const FAILED_HEALTH_CHECKS_THRESHOLD = 5;
+const HEALTH_CHECK_INTERVAL_MS = 1000 * 30;   // interval on which to run health checks
+
+/**
+*   Class that manages and communicates with the authority registry. Maintains a list of all
+*   authority servers.
+*/
+export class AuthorityRegistry {
+  constructor () {
+    this.ip = ip.address();
+    this.port = parseInt(process.env.PORT);
+
+    const ipAndPort = `${this.ip}:${this.port}`;
+    this.fullUrl = (this.port === HTTPS_PORT) ? `https://${ipAndPort}` : `http://${ipAndPort}`;
```

Comment:
I think here it would maybe be better to specify connection protocol in separate env variable instead of trying to be clever about it and switch depending on port user specified.